File: C#/Wall.cs

namespace Master_Thesis;

class Wall
{
    // Geometrical and physical parameters of each instance of the class Wall.
    private double _centreYCor;
    private double _centreZCor;

    const double tWall = 0.3;
    const double heightOfBuilding = 12;
    const double youngModulus = 33000000000;
    private const double ShearCorrectionFactor = 0.8333333333;

    // Constructing an instance of the class Wall and calculating its geometrical and physical parameters
    public Wall(Point a, Point b)
    {
        CalculateParameters(a, b);
    }

    private void CalculateParameters(Point a, Point b)
    {
        // Assigning parameters of a wall
        var gModulus = youngModulus / (2 * (1 + 0.2));
        
        // Calling coordinates of the wall's beginning and end from the Point class
        var startYCor = a.DisplayCoordinates()[0];
        var startZCor = a.DisplayCoordinates()[1];
        var endYCor = b.DisplayCoordinates()[0];
        var endZCor = b.DisplayCoordinates()[1];

        // Finding the orientation of the wall's main axis
        if (startYCor - endYCor == 0)
            Orientation = 'z';
        else
            Orientation = 'y';

        // Coordinates of the wall's centre
        _centreYCor = startYCor + (endYCor - startYCor) / 2;
        _centreZCor = startZCor + (endZCor - startZCor) / 2;

        // Length of the wall
        if (Orientation == 'y')
            Length = (endYCor - startYCor);

        else if (Orientation == 'z')
            Length = (endZCor - startZCor);
        else
            throw new Exception("Invalid direction of the wall");

        // Second Moment of Inertia
        double momentOfInertia;
        if (Orientation == 'y')
        {
            momentOfInertia = tWall * ((endYCor - startYCor) * (endYCor - startYCor) * (endYCor - startYCor)) / 12;
        }
        else if (Orientation == 'z')
        {
            momentOfInertia = tWall * ((endZCor - startZCor) * (endZCor - startZCor) * (endZCor - startZCor)) / 12;
        }
        else
        {
            throw new Exception("Invalid direction of the wall");
        }

        InertiaMoment = momentOfInertia;

        // Bending Stiffness
        var bendingStiffness = (3 * youngModulus * momentOfInertia) / (heightOfBuilding * heightOfBuilding * heightOfBuilding);

        // Shear Area
        double shearArea;
        if (Orientation == 'y')
            shearArea = (endYCor - startYCor) * tWall * ShearCorrectionFactor;

        else if (Orientation == 'z')
            shearArea = (endZCor - startZCor) * tWall * ShearCorrectionFactor;

        else
            throw new Exception("Invalid direction of the wall");
        ShearArea = shearArea;

        // Shear Stiffness
        var shearStiffness = (gModulus * shearArea) / heightOfBuilding;

        // Equivalent Stiffness
        EquivalentStiffness = (bendingStiffness * shearStiffness) / (bendingStiffness + shearStiffness);

        // Product of centre coordinates and equivalent stiffness
        if (Orientation == 'y')
            CorStiffProduct = EquivalentStiffness * _centreZCor;

        else if (Orientation == 'z')
            CorStiffProduct = EquivalentStiffness * _centreYCor;

        else
            throw new Exception("Invalid direction of the wall");

        // Torsional Moment of Inertia
        TorsionalMoment = 0.323 * Length * tWall * tWall * tWall;
    }

    // Following methods are used to call each parameters of a wall separately
    public double EquivalentStiffness { get; private set; }
    public char Orientation { get; private set; }
    public double CorStiffProduct { get; private set; }
    public double Length { get; private set; }
    public double TorsionalMoment { get; private set; }
    public double InertiaMoment { get; private set; }
    public double ShearArea { get; private set; }

    public double[] CentreCoordinates()
    {
        double[] shearCentreCoordinates = new double[2];
        shearCentreCoordinates[0] = _centreYCor;
        shearCentreCoordinates[1] = _centreZCor;
        return shearCentreCoordinates;
    }

    
}

File: Python/Wall.py
from Point import Point


class Wall:

    shearCorrectionFactor = 0.8333333333
    E = 33000000000
    heightOfBuilding = 60

    # Creating an instance of the class Wall and calculating its geometrical and physical properties
    def __init__(self, A: Point, B: Point, tWall):

        self.startYCor = A.DisplayCoordinates()[0]
        self.startZCor = A.DisplayCoordinates()[1]
        self.endYCor = B.DisplayCoordinates()[0]
        self.endZCor = B.DisplayCoordinates()[1]
        self.CalculateParameters(tWall)

    # Calling coordinates of the wall's start and end
    def StartEndCord(self):
        startEndCord = [self.startYCor, self.startZCor, self.endYCor, self.endZCor]
        return startEndCord

    # Main function calculating parameters
    def CalculateParameters(self, tWall):

        # Thickness of wall
        self.thickness = tWall

        # Assigning parameters of a wall
        gModulus = self.E / (2 * (1 + 0.2))

        # Finding the orientation of the wall's main axis
        if self.startYCor - self.endYCor == 0:
            self.orientation = 'z'
        else:
            self.orientation = 'y'

        # Coordinates of the wall's centre
        self.centreYCor = self.startYCor + (self.endYCor - self.startYCor) / 2
        self.centreZCor = self.startZCor + (self.endZCor - self.startZCor) / 2

        # Length of the wall
        if self.orientation == 'y':
             self.length = (self.endYCor - self.startYCor)

        elif self.orientation == 'z':
             self.length = (self.endZCor - self.startZCor)

        # Second Moment of Inertia
        if self.orientation == 'y':
             self.momentOfInertia = self.thickness * (
                     (self.endYCor - self.startYCor) * (self.endYCor - self.startYCor) * (self.endYCor - self.startYCor)) / 12
        elif self.orientation == 'z':
             self.momentOfInertia = self.thickness * (
                     (self.endZCor - self.startZCor) * (self.endZCor - self.startZCor) * (self.endZCor - self.startZCor)) / 12

        # Bending Stiffness
        self.bendingStiffness = (3 * self.E * self.momentOfInertia) / (
                 self.heightOfBuilding * self.heightOfBuilding * self.heightOfBuilding)

        # Shear Area
        if self.orientation == 'y':
            self.shearArea = (self.endYCor - self.startYCor) * self.thickness * self.shearCorrectionFactor
        elif self.orientation == 'z':
            self.shearArea = (self.endZCor - self.startZCor) * self.thickness * self.shearCorrectionFactor

        # Shear Stiffness
        self.shearStiffness = (gModulus * self.shearArea) / self.heightOfBuilding

        # Equivalent Stiffness
        self.equivalentStiffness = (self.bendingStiffness * self.shearStiffness) / (self.bendingStiffness + self.shearStiffness)

        # Product of the quivalent stiffness and centre's coordinates
        if self.orientation == 'y':
             self.corStiffProduct = self.equivalentStiffness * self.centreZCor

        elif self.orientation == 'z':
             self.corStiffProduct = self.equivalentStiffness * self.centreYCor

        # Torsional Moment of Inertia
        self.torsionalMoment = 0.323 * self.length * self.thickness ** 3


    # Coordinates of wall's centre
    def CentreCoordinates(self):
        shearCentreCoordinates=[self.centreYCor, self.centreZCor]
        return shearCentreCoordinates
